fuel-vm/src/error.rs

```diff
@@ -38,6 +39,9 @@ pub enum InterpreterError<StorageError> {
     /// The provided transaction isn't valid.
     #[display(fmt = "Failed to validate the transaction: {_0}")]
     TransactionValidity(ValidityError),
+    /// Failed while checking the transaction.
+    #[display(fmt = "Failed to check the transaction: {_0:?}")]
+    CheckError(CheckError),
```

Comment:
Hmm, `CheckError` already contains `ValidityError` inside. Maybe it makes sense to move a new error to `ValidityError`. In this case, you don't need to add `CheckError` here

Reply:
It would make more sense to remove the `ValidityError` variant because there is a `ValidityError` variant of `CheckError`.

Reply:
Okay. I reread what you said.

Maybe `into_ready` could return a `ValidityError`? Is that what you're saying. I had to add this because we call `into_ready` inside the `Transactor`. TBH, I don't know why we are cluttering the `InterpreterError`s with a test harnesses stuff, like `Transactor`.

I'd almost say that `Transactor` should have its own error that can wrap `CheckError` and `InterpreterError`. The whole caught error thing in `Transactor` is kinda ugly IMO, but it's kinda out of scope right now.